wandb/sdk/internal/handler.py

```diff
@@ -454,16 +484,46 @@ def handle_history(self, record: Record) -> None:
         # Inject _runtime if it is not present
         if history_dict is not None:
             if "_runtime" not in history_dict:
-                self._history_assign_runtime(record, history_dict)
+                self._history_assign_runtime(record.history, history_dict)
 
-        self._history_update(record, history_dict)
+        self._history_update(record.history, history_dict)
         self._dispatch_record(record)
-        self._save_history(record)
-
+        self._save_history(record.history)
         updated = self._update_summary(history_dict)
         if updated:
             self._save_summary(self._consolidated_summary)
 
+    def _flush_partial_history(self, record: Record) -> None:
+        if self._partial_history:
+            # Inject _runtime if it is not present
+            if "_runtime" not in self._partial_history:
+                self._history_assign_runtime(
+                    record.request.partial_history, self._partial_history
+                )
+
+            for k, v in self._partial_history.items():
+                item = record.request.partial_history.item.add()
+                item.key = k
+                item.value_json = json.dumps(v)
+
+            self._history_update(record.request.partial_history, self._partial_history)
+            self._dispatch_record(record)
```

Comment:
I think you should dispatch a HistoryRecord and leave sender untouched with respect to partial history?

Reply:
Yeah... I don't remember whey I did it this way... but in theory it seems like the cleaner way to do it